app/src/main/res/layout/fragment_nutrition_product.xml

```diff
@@ -53,21 +53,37 @@
             android:padding="@dimen/spacing_small"
             android:textIsSelectable="true"
             android:textSize="@dimen/font_normal" />
+     <LinearLayout   android:layout_width="fill_parent"
+         android:layout_height="wrap_content"
+         android:layout_marginTop="@dimen/spacing_tiny"
+         android:layout_marginBottom="@dimen/spacing_tiny"
+         android:layout_marginLeft="@dimen/spacing_normal"
+         android:layout_marginRight="@dimen/spacing_normal"
+         android:padding="@dimen/spacing_small"
+         android:background="@drawable/textview_full">
 
-        <TextView
-            android:id="@+id/textNutrientTxt"
-            android:layout_width="fill_parent"
-            android:layout_height="wrap_content"
-            android:layout_marginLeft="@dimen/spacing_normal"
-            android:layout_marginRight="@dimen/spacing_normal"
-            android:layout_marginTop="@dimen/spacing_tiny"
-            android:background="@drawable/textview_full"
-            android:padding="@dimen/spacing_small"
-            android:text="@string/txtNutrientLevel100g"
-            android:textIsSelectable="true"
-            android:textSize="@dimen/font_normal"
-            android:textStyle="bold" />
 
+         <TextView
+             android:id="@+id/textNutrientTxt"
+             android:layout_width="wrap_content"
+             android:layout_height="wrap_content"
+             android:text="@string/txtNutrientLevel100g"
+             android:textIsSelectable="true"
+             android:layout_marginRight="@dimen/spacing_normal"
+             android:textSize="@dimen/font_normal"
+             android:textStyle="bold" />
+         <TextView
+             android:id="@+id/txtNoDataString"
+             android:layout_width="wrap_content"
+             android:visibility="gone"
+             android:layout_height="wrap_content"
+             android:layout_marginRight="@dimen/spacing_normal"
+             android:layout_marginLeft="@dimen/spacing_normal"
+             android:text="@string/txtNoData"
+             android:textIsSelectable="true"
+             android:textSize="@dimen/font_normal"
+             android:textStyle="bold" />
+     </LinearLayout>
```

Comment:
I don't think it's a good idea to add a text view for no data case at fragment level. 

It's better to tweak the view binding process in `NutrientLevelListAdapter`. Considering the no data as one of the case and make your item view more resistant.

Reply:
I don't know how modifying the list adapter will help, If we want the "No data" text with the header.The other approach which I came up is to append "No data" text to the header in the NutritionProductFragment class.